troi/patches/top_missed_recordings_for_year.py

```diff
@@ -0,0 +1,159 @@
+from datetime import datetime
+
+import psycopg2
+import psycopg2.extras
+
+import troi
+from troi import Recording, Element
+from troi.playlist import PlaylistShuffleElement, PlaylistRedundancyReducerElement
+from troi.musicbrainz.recording_lookup import RecordingLookupElement
+
+
+
+class MissedRecordingsElement(Element):
+
+    """ This element looks up top tracks for 3 users minus the tracks of one given user to form 
+        the core of the missed tracks playlist for yim. """
+
+    def __init__(self, user_id, similar_user_ids, db_connect_str):
+        Element.__init__(self)
+        self.user_id = user_id
+        self.similar_user_ids = similar_user_ids
+        self.db_connect_str = db_connect_str
+
+    @staticmethod
+    def inputs():
+        return [ ]
+
+    @staticmethod
+    def outputs():
+        return [ Recording ]
+
+    def read(self, inputs):
+
+        output = []
+        with psycopg2.connect(self.db_connect_str) as conn:
+            with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as curs:
+ 
+                query = """WITH exclude_tracks AS (
```

Comment:
Can remove the extra fields from the query because only using recording_mbid from results

Reply:
Oh, the listen_count should be written to the LB object. Both fixed.